framework/include/actions/Action.h

```diff
@@ -44,8 +46,48 @@ class Action : public ConsoleStreamInterface
 
   virtual void act() = 0;
 
+  /**
+   * The name of the action
+   *
+   * When using MOOSE input files, this is the last portion
+   * of the full input syntax:
+   *    [System]
+   *      [./name]
+   *
+   *    [System]
+   *      [./Subsystem]
+   *        [./name]
+   */
   const std::string & name() const { return _name; }
 
+  ///@{
+  /**
+   * The full input file and base syntax for the action
+   *
+   * _full_syntax = System/name
+   * _base_syntax = System
+   * [System]
+   *   [./name]
+   *
+   * _full_syntax = System/Subsystem/name
+   * _base_syntax = System/Subsystem
+   * [System]
+   *   [./Subsystem]
+   *     [./name]
+   */
+  const std::string & getInputSyntax() const { return _full_syntax; }
```

Comment:
Well I still argue that long name is slightly better than InputfileSyntax. The reason is that I can set a long name outside of the input file. In fact I believe we do this somewhere in the modules.